src/main/java/io/gravitee/policy/groovy/model/http/ScriptableHttpRequest.java

```diff
@@ -0,0 +1,217 @@
+import io.gravitee.common.http.HttpVersion;
+import io.gravitee.common.util.MultiValueMap;
+import io.gravitee.gateway.api.buffer.Buffer;
+import io.gravitee.gateway.api.http.HttpHeaders;
+import io.gravitee.gateway.reactive.api.context.GenericRequest;
+import io.gravitee.gateway.reactive.api.context.HttpRequest;
+import io.gravitee.policy.groovy.model.ScriptableHttpHeaders;
+import javax.net.ssl.SSLSession;
+import lombok.Getter;
+
+/**
+ * @author Antoine CORDIER (antoine.cordier at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class ScriptableHttpRequest implements GenericRequest {
+
+    @Getter
+    private final HttpRequest request;
+
+    @Getter
+    private final String content;
+
+    public ScriptableHttpRequest(HttpRequest request, Buffer buffer) {
+        this.request = request;
+        this.content = buffer.toString();
+    }
+
+    public ScriptableHttpRequest(HttpRequest request) {
+        this(request, Buffer.buffer());
+    }
+
+    @Override
+    public String id() {
+        return request.id();
+    }
+
+    @Override
+    public String transactionId() {
+        return request.transactionId();
+    }
+
+    public String getTransactionId() {
+        return transactionId();
+    }
+
+    @Override
+    public String clientIdentifier() {
+        return request.clientIdentifier();
+    }
+
+    @Override
+    public String uri() {
+        return request.uri();
+    }
+
+    public String getUri() {
+        return uri();
+    }
+
+    @Override
+    public String host() {
+        return request.host();
+    }
+
+    public String getHost() {
+        return host();
+    }
+
+    @Override
+    public String originalHost() {
+        return request.originalHost();
+    }
+
+    public String getOriginalHost() {
+        return originalHost();
+    }
+
+    @Override
+    public String path() {
+        return request.path();
+    }
+
+    public String getPath() {
+        return path();
+    }
+
+    @Override
+    public String pathInfo() {
+        return request.pathInfo();
+    }
+
+    public String getPathInfo() {
+        return pathInfo();
+    }
+
+    @Override
+    public String contextPath() {
+        return request.contextPath();
+    }
+
+    public String getContextPath() {
+        return contextPath();
+    }
+
+    @Override
+    public MultiValueMap<String, String> parameters() {
+        return request.parameters();
+    }
+
+    public MultiValueMap<String, String> getParameters() {
+        return parameters();
+    }
+
+    @Override
+    public MultiValueMap<String, String> pathParameters() {
+        return request.pathParameters();
+    }
+
+    public MultiValueMap<String, String> getPathParameters() {
+        return pathParameters();
+    }
+
+    @Override
+    public HttpHeaders headers() {
+        return request.headers();
+    }
+
+    public ScriptableHttpHeaders getHeaders() {
+        return new ScriptableHttpHeaders(request.headers());
```

Comment:
Does `headers() `should also return a wrapper?

Reply:
I don't think we can because the Map and HttpHeaders interface have type collide